packages/scenario/src/answer/ExpectedApproximateUOMAnswer.ts

```diff
@@ -0,0 +1,168 @@
+		 * Specifies the acceptable difference between the expected value and an
+		 * actual value compared in an assertion.
+		 *
+		 * This is semantically equivalent to the `error` parameter in
+		 * {@link https://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/number/IsCloseTo.html | Hamcrest's `isCloseTo`}
+		 * assertion comparison. It notably differs from the `numDigits` parameter
+		 * in
+		 * {@link https://vitest.dev/api/expect.html#tobecloseto | Vitest's `toBeCloseTo`},
+		 * which is less clear in its precision-of-comparison semantics.
+		 *
+		 * **PORTING NOTES**
+		 *
+		 * The intent here is to port JavaRosa tests with `closeTo` calls with high
+		 * confidence that they have the same error tolerance (notwithstanding any
+		 * potential discrepancy in their respective floating point implementations;
+		 * such discrepancies would be unexpected, given both implement IEEE-754).
+		 */
+		readonly errorTolerance: Value | number
+	) {
+		super();
+
+		this.stringValue = String(uomValue.unitValue);
+	}
+
+	override inspectValue(): JSONValue {
+		return this.uomValue.inspectValue(this.errorTolerance);
+	}
+
+	/**
+	 * @todo Given that several comparisons ported form JavaRosa fail with a
+	 * difference exceedingly close to the provided {@link errorTolerance},
+	 * consider whether those failures are rooted in this implementation. However,
+	 * a quick glance at the
+	 * {@link https://github.com/hamcrest/JavaHamcrest/blob/master/hamcrest/src/main/java/org/hamcrest/number/IsCloseTo.java | equivalent Hamcrest logic}
+	 * suggests that our logic is basically the same. (And a quick spike to use
+	 * the exact same formulation of the same logic produces the same result.)
+	 */
+	isCloseTo(actual: ComparableAnswer): boolean {
+		const actualValue = this.uomValue.constructor.fromAnswer(actual).unitValue;
+
+		if (Number.isNaN(actualValue)) {
+			return false;
+		}
+
+		const { uomValue, errorTolerance } = this;
+		const expectedValue = uomValue.unitValue;
+
+		if (expectedValue === actualValue) {
+			return true;
+		}
+
+		const difference = Math.abs(actualValue - expectedValue);
+
+		return difference <= Number(errorTolerance);
+	}
+}
+
+type SquareMeter = number;
+
+class AreaValue extends UOMValue {
+	readonly uom = 'm^2';
+}
+
+/**
+ * @see {@link ExpectedApproximateUOMAnswer.errorTolerance}
+ */
+export const expectedArea = (
+	area: SquareMeter,
+	errorTolerance: number
+): ExpectedApproximateUOMAnswer => {
+	const value = new AreaValue(area);
+
+	return new ExpectedApproximateUOMAnswer(value, errorTolerance);
+};
+
+type Meter = number;
+
+class DistanceValue extends UOMValue {
+	readonly uom = 'm';
+}
+
+/**
+ * @see {@link ExpectedApproximateUOMAnswer.errorTolerance}
+ */
+export const expectedDistance = (
+	distance: Meter,
+	errorTolerance: number
+): ExpectedApproximateUOMAnswer => {
+	const value = new DistanceValue(distance);
+
+	return new ExpectedApproximateUOMAnswer(value, errorTolerance);
+};
+
+type FractionalDay = number;
```

Comment:
Since TypeScript (mostly) doesn't have nominal types, the effective type is `number`. This type is intended to serve as a bit of additional inline documentation. I generally find it helpful, especially later on during maintenance or code archaeology or whatever, to have some explicit clarification of a value's UOM when represented as a bare numeric value. What would be the benefit of the bare `number` type?